packages/toolpad-app/src/toolpad/AppEditor/AppEditorShell.tsx

```diff
@@ -130,42 +129,37 @@ export default function AppEditorShell({ appId, ...props }: ToolpadShellProps) {
   const hasUnsavedChanges = domLoader.unsavedChanges > 0;
   const isSaving = domLoader.saving;
 
+  const savedStatusIcon = hasUnsavedChanges ? <CloudSyncIcon /> : <CloudDoneIcon />;
+
   return (
     <ToolpadShell
       actions={
         <Stack direction="row" gap={1} alignItems="center">
-          <Tooltip title={getSaveStateMessage(isSaving, hasUnsavedChanges)}>
-            <Box display="flex" flexDirection="row" alignItems="center" mr={1}>
-              {isSaving ? (
-                <CircularProgress size={16} color="inherit" sx={{ mr: 1 }} />
-              ) : (
-                <CloudDoneIcon
-                  color={hasUnsavedChanges ? 'disabled' : 'success'}
-                  fontSize="medium"
-                />
-              )}
-            </Box>
-          </Tooltip>
           <Button
             variant="outlined"
-            color="inherit"
+            color="primary"
             component="a"
             href={`/app/${appId}/preview`}
             target="_blank"
-            endIcon={<OpenInNewIcon />}
           >
             Preview
           </Button>
-          <Button
-            variant="outlined"
-            color="inherit"
-            onClick={handleCreateReleaseDialogOpen}
-            endIcon={<RocketLaunchIcon />}
-          >
+          <Button variant="outlined" color="primary" onClick={handleCreateReleaseDialogOpen}>
             Deploy
           </Button>
         </Stack>
       }
+      status={
+        <Tooltip title={getSaveStateMessage(isSaving, hasUnsavedChanges)}>
+          <Box sx={{ display: 'flex', flexDirection: 'row', alignItems: 'center', mr: 3 }}>
+            {isSaving ? (
```

Comment:
This isn't really about these specific changes, but just to maybe discuss: Could we assume that if there are unsaved changes the app is already saving, and show the loader?

In which scenarios could there be unsaved changes, but the app isn't saving? And what could the user do about it in those cases not to lose progress?
I though about that already when I made changes to this, but I wasn't sure.

Reply:
> Could we assume that if there are unsaved changes the app is already saving, and show the loader?

it's a three state:

1. all progress saved
2. local changes, not synced
3. syncing

I don't see what would be the problem with visualizing the three states, and what we would gain by merging 2. and 3.

> In which scenarios could there be unsaved changes, but the app isn't saving?

It's an edge-case, but maybe: make a change, then undo before the change was synced to the server

Reply:
I was just wondering if the icon changes could be confusing for the user, since most users might only care about whether the app is saved or saving - and get worried if it's neither saved or saving, especially if there's no way to save manually.
But it should be fine to keep this as is and keep it if it never causes any real issues - and the current behavior might work better with future changes. Just wanted to suggest the idea just in case.

Reply:
Updated

Reply:
I want to make sure I understand this convo — is the question here if it is valuable to show 3 icons (or other indicators) for the 3 states? 
🟢 All changes saved & synced
🟡 Saved changes not synced
🔴 Unsaved changes not synced

Could another case for 2 🟡 be an interrupted internet connection for the theoretical cloud version? Could this potentially happen in the self hosted version? 

----

On a related note, I'm not a huge fan of the icon that material has for cloud sync vs cloud done since it scales the cloud down a lot. I was also looking at just the regular sync icon instead — it's even more of a visual change but at least a similar scale. 
<img width="59" alt="Sync" src="https://user-images.githubusercontent.com/106991024/189974455-59b421b3-b605-403b-bec4-9b3df6f53e01.png">

Then there's both a Sync Disabled or Sync Problem icon to indicate an issue like what number 2 describes. 
<img width="68" alt="Sync Disabled" src="https://user-images.githubusercontent.com/106991024/189974479-333c6f90-f138-4496-8dc9-af8bf6768d82.png"> <img width="77" alt="Sync Problem" src="https://user-images.githubusercontent.com/106991024/189974720-a9c632e4-7232-4dfd-b88d-6a27333909e6.png">

Reply:
> is the question here if it is valuable to show 3 icons

No, the conversation is about

1. all changes synced
2. there are local changes ready to be synced
3. actually busy syncing changes

> Could another case for 2 🟡 be an interrupted internet connection for the theoretical cloud version?

Good point, that's an additional state we should visualize.

> Could this potentially happen in the self hosted version?

Yes, it represents saving to the backend